### ISSUE TITLE
Not able to reproduce the diversity in the tf-bind8 task

### ISSUE THREAD

**Issue by charlesxu90** [MJ10/BioSeq-GFN-AL#6]
Dear authors, 

I tried to reproduce the results using the code provided. However, I found the results different from the results you reported in the paper.

Below is what I got for the tf-bind8 task. The diversity seems very different. Besides, the novelty metric is not provided in this repository. Could you provide the code for such calculation?

```shell
Generating samples
top 100, scores 0.8414675438404083, max score 0.8704386949539185, 50 pl score 0.8275883197784424, diversity 1.7107070707070706
top 128, scores 0.8304508188739419, max score 0.8704386949539185, 50 pl score 0.8275883197784424, diversity 2.442667322834646
top 1000, scores 0.8304508188739419, max score 0.8704386949539185, 50 pl score 0.8275883197784424, diversity 2.442667322834646

```

**Comment by MJ10** [comment]
Hi, could you please share the exact command you ran?

**Comment by charlesxu90** [comment]
Below is the code I used for running.
```shell
python run.py --task tfbind --save_path results/tfbind/test_mlp.pkl.gz \
  --vocab_size 4 --max_len 8 \
  --num_rounds 1 --num_sampled_per_round 128 \
  --proxy_learning_rate 1e-4 \
  --proxy_num_hid 64 --proxy_L2 1e-4 --proxy_num_iterations 10000 \
  --n_sample_round 100 \
  --gen_do_explicit_Z 1 --acq_fn ucb --gen_num_iterations 2500 --gen_reward_exp 8 --gen_data_sample_per_step 8 \
  --gen_Z_learning_rate 1e-1 --gen_learning_rate 1e-3
```

The result from each run seems to be different, below is the latest result from a run.
```shell
Generating samples
top 100, scores 0.8689483880996705, max score 0.9171133637428284, 50 pl score 0.8669236898422241, diversity 0.3511111111111111
top 128, scores 0.868505485355854, max score 0.9171133637428284, 50 pl score 0.8669236898422241, diversity 0.27583661417322836
top 1000, scores 0.868505485355854, max score 0.9171133637428284, 50 pl score 0.8669236898422241, diversity 0.27583661417322836

```

**Comment by charlesxu90** [comment]
BTW, the run.py does the samething as run_tfbind.py. I'm playing with the code and trying to prune the code to make it easy to read. 

My pruned codes are shared on [this folk](https://github.com/charlesxu90/BioSeq-GFN-AL).

**Comment by MJ10** [comment]
Apologies for the late reply! I am unable to reproduce these results. I get $3.96+0.52$ when I run on 5 random seeds. Have you tried running multiple seeds?
